packages/sg721/src/lib.rs

```diff
@@ -51,27 +91,129 @@ pub enum ExecuteMsg<T, E> {
         /// Metadata JSON Schema
         token_uri: Option<String>,
         /// Any custom extension used by this contract
-        extension: T,
+        extension: TNftExtensionMsg,
     },
+
     /// Burn an NFT the sender has access to
     Burn {
         token_id: String,
     },
-    /// Extension msg
+
+    /// Custom msg execution. This is a no-op in default implementation.
     Extension {
-        msg: E,
+        msg: TExtensionMsg,
     },
-    /// Update specific collection info fields
-    UpdateCollectionInfo {
-        collection_info: UpdateCollectionInfoMsg<RoyaltyInfoResponse>,
+
+    /// The creator is the only one eligible to update NFT's token uri and onchain metadata (`NftInfo.extension`).
+    /// NOTE: approvals and owner are not affected by this call, since they belong to the NFT owner.
+    UpdateNftInfo {
+        token_id: String,
+        token_uri: Option<String>,
+        extension: TNftExtensionMsg,
+    },
+
+    /// Sets address to send withdrawn fees to. Only owner can call this.
+    SetWithdrawAddress {
+        address: String,
+    },
+    /// Removes the withdraw address, so fees are sent to the contract. Only owner can call this.
+    RemoveWithdrawAddress {},
+    /// Withdraw from the contract to the given address. Anyone can call this,
+    /// which is okay since withdraw address has been set by owner.
+    WithdrawFunds {
+        amount: Coin,
     },
-    /// Called by the minter to update trading start time
-    UpdateStartTradingTime(Option<Timestamp>),
-    // Freeze collection info from further updates
-    FreezeCollectionInfo,
+}
+
+impl<TNftExtensionMsg, TCollectionExtensionMsg, TExtensionMsg>
```

Comment:
Is lib.rs the right place for this?

Reply:
100%, before PR changes `sg721` isnt suppose to hold much specific data or logic - since everything is in `cw721`. so lib.rs in here just binds cw721 and adds only few additional sg specific data.